src/photon/photon_scheduler.c

```diff
@@ -500,6 +669,13 @@ int main(int argc, char *argv[]) {
   if (!node_ip_address) {
     LOG_FATAL("please specify the node IP address with -h switch");
   }
+  int num_workers = 0;
+  if (num_workers_str) {
+    num_workers = strtol(num_workers_str, NULL, 10);
```

Comment:
We should check that `strtol` succeeded and that `num_workers_str` was not malformed